### PR TITLE
refactor(localnet): reduce number of docker compose files

### PR DESCRIPTION
**Pull request by gartnera**
# Description

Reduce number of docker compose files by instead injecting environment variables where possible. Use these environment variables in conditionals rather than command line arguments also.

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [x] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

## Summary by CodeRabbit

- **Chores**
  - Streamlined build and test workflows to improve efficiency.
  - Updated environment variables and configurations in various local network setup files.
  - Modified scripts to use new environment variables, enhancing flexibility and maintainability.

### CODE REVIEW

File: .github/workflows/execute_advanced_tests.yaml [hunk 1]
```diff
@@ -137,5 +134,4 @@ jobs:
       - name: Full Log Dump On Failure
         if: failure()
         run: |
-          cd contrib/localnet
-          docker compose logs
\ No newline at end of file
+          make stop-localnet
```
Comment (swift1337) [inline .github/workflows/execute_advanced_tests.yaml:137]: nit: add a new line

File: Makefile [hunk 2]
```diff
@@ -242,24 +250,30 @@ zetanode-upgrade: zetanode
...
-stop-localnet:
 start-e2e-import-mainnet-test: zetanode
 	@echo "--> Starting e2e import-data test"
-	cd contrib/localnet/ && ./scripts/import-data.sh mainnet && $(DOCKER) compose -f docker-compose.yml -f docker-compose-import-data.yml up -d
+	export ZETACORED_IMPORT_GENESIS_DATA=true && \
+	export ZETACORED_START_PERIOD=10m && \
+	cd contrib/localnet/ && ./scripts/import-data.sh mainnet && $(DOCKER) compose -f docker-compose.yml up -d
 
-stop-test:
+stop-localnet:
```
Comment (swift1337) [inline Makefile:276]: nice

File: contrib/localnet/scripts/import-data.sh [hunk 3]
```diff
@@ -7,9 +7,9 @@ fi
...
-rm -rf ~/genesis_export/
-mkdir ~/genesis_export/
+rm -rf ~/.zetacored/genesis_data
+mkdir ~/.zetacored/genesis_data
 echo "Download Latest State Export"
 LATEST_EXPORT_URL=$(curl https://snapshots.zetachain.com/latest-state-export | jq -r ."${NETWORK}")
 echo "LATEST EXPORT URL: ${LATEST_EXPORT_URL}"
-wget -q ${LATEST_EXPORT_URL} -O ~/genesis_export/exported-genesis.json
\ No newline at end of file
+wget -q ${LATEST_EXPORT_URL} -O ~/.zetacored/genesis_data/exported-genesis.json
```
Comment (swift1337) [inline contrib/localnet/scripts/import-data.sh:15]: nit: add a new line

**Review by swift1337 — APPROVED** [review]
LGTM; Small note: can you briefly describe new env vars in `LOCAL_TESTING.md`?

**Review by lumtis — APPROVED** [review]

**Review by kingpinXD — APPROVED** [review]

**Review by CharlieMc0 — APPROVED** [review]

**Pull request merged**